admin-ui/plugins/auth-server/redux/sagas/authNLdapSaga.js

```diff
@@ -0,0 +1,118 @@
+import { addLdapSuccess, addLdapFailure, editLdapSuccess } from '../features/authNLdapSlice'
+import { updateToast } from 'Redux/features/toastSlice'
+import { DatabaseLDAPConfigurationApi } from 'jans_config_api'
+import { initAudit } from '@/redux/sagas/SagaUtils'
+import { addAdditionalData } from '@/utils/TokenController'
+import { DELETION, FETCH, UPDATE } from '@/audit/UserActionType'
+import { postUserAction } from '@/redux/api/backend-api'
+import { LDAP } from '@/utils/ApiResources'
+
+function deleteLdapApi(configId) {
+  const api = new DatabaseLDAPConfigurationApi()
+  return new Promise((resolve, reject) => {
+    api.deleteConfigDatabaseLdapByName(configId, (error, data) => {
+      if (error) reject(error)
+      else resolve(data)
+    })
+  })
+}
+
+function* deleteLdapSaga({ payload }) {
+  const audit = yield* initAudit()
+  console.log('action in delete ldap saga', payload)
+  try {
+    addAdditionalData(audit, DELETION, LDAP, { message: payload?.userMessage })
+    yield call(deleteLdapApi, payload?.configId)
+    yield put(updateToast(true, 'success', 'LDAP deleted successfully'))
+    yield put(deleteLdapSuccess())
+    yield call(postUserAction, audit)
+    yield put({ type: 'authNLdap/getLdapList' })
+  } catch (e) {
+    yield put(updateToast(true, 'error', 'Error deleting LDAP'))
+    yield put(deleteLdapFailure())
+  }
+}
+function addLdapApi(payload) {
+  const api = new DatabaseLDAPConfigurationApi()
+  return new Promise((resolve, reject) => {
+    api.postConfigDatabaseLdap(payload, (error, data) => {
+      if (error) reject(error)
+      else resolve(data)
+    })
+  })
+}
+
+function* addLdapSaga(action) {
+  const audit = yield* initAudit()
+  try {
+    console.log('action in add ldap saga', action)
+
+    addAdditionalData(audit, FETCH, LDAP, { message: action?.payload?.action_message })
+    yield call(addLdapApi, action.payload)
+    yield put(updateToast(true, 'success', 'LDAP added successfully'))
+    yield put(addLdapSuccess())
+    yield call(postUserAction, audit)
+    action.onSuccessApply?.()
+  } catch (e) {
+    yield put(updateToast(true, 'error', 'Error adding LDAP'))
+    yield put(addLdapFailure())
+  }
+}
+
+function editLdapApi(payload) {
+  const api = new DatabaseLDAPConfigurationApi()
+  return new Promise((resolve, reject) => {
+    api.putConfigDatabaseLdap(payload, (error, data) => {
+      if (error) reject(error)
+      else resolve(data)
+    })
+  })
+}
+
+function* editLdapSaga(action) {
+  const audit = yield* initAudit()
+  try {
+    addAdditionalData(audit, UPDATE, LDAP, { message: action?.payload?.action_message })
+    yield call(editLdapApi, action?.payload)
+    yield put(updateToast(true, 'success', 'LDAP updated successfully'))
+    yield put(editLdapSuccess())
+    action?.onSuccessApply?.()
+    yield call(postUserAction, audit)
+  } catch (e) {
+    yield put(updateToast(true, 'error', 'Error updating LDAP'))
+    yield put(addLdapFailure())
```

Comment:
Incorrect error action dispatched in editLdapSaga

The `editLdapSaga` dispatches `addLdapFailure()` on error instead of `editLdapFailure()`.


```diff
-    yield put(addLdapFailure())
+    yield put(editLdapFailure())
```

Note: You'll need to import `editLdapFailure` from the slice if it exists, or create it if it doesn't.